src/errors.rs

```diff
@@ -0,0 +1,73 @@
+use crate::de;
+use crate::ser;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
```

Comment:
I'd probably add a `#[non_exhaustive]` here to ensure that adding new error variants remains a non-API-breaking operation. It's available since Rust version 1.40.

Reply:
Very good suggestion, thanks!